core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -395,6 +415,61 @@ object ConsumerGroupCommand extends Logging {
       result
     }
 
+    def deleteOffsets(): (Option[Throwable], Map[TopicPartition, Throwable]) = {
+      val groupId = opts.options.valueOf(opts.groupOpt)
+      val topics = opts.options.valuesOf(opts.topicOpt)
+
+      val (topicWithPartitions, topicWithoutPartitions) = topics.asScala.partition(_.contains(":"))
+
+      val knownPartitions = topicWithPartitions.flatMap { topicArg =>
+        val split = topicArg.split(":")
+        split(1).split(",").map { partition =>
+          new TopicPartition(split(0), partition.toInt)
+        }
+      }
+
+      val describeTopicsResult = adminClient.describeTopics(
```

Comment:
Hmm.. I noticed that the OffsetDelete API does not check topic existence, which is a little inconsistent with OffsetCommit. If we added existence checks, then I think we could skip this call to `describeTopics`.

Reply:
OffsetDelete API does check topic existence if I remember correctly.

The describe topic here is not used to check the existence but to get the partitions of a topic if not provided. Topics with partitions go directly to the next step and rely and the existence check of the Offset Delete API.

Reply:
Makes sense. I reviewed a bit too hastily. However, can you double-check OffsetDelete handling logic? I don't see topic existence checks.

Reply:
The check is here: https://github.com/apache/kafka/blob/70d1bb40d958918ac691ba906a35c4a158815d78/core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala#L569

It checks that the group has a commit for the topic/partition. if not, it considers that the topic/partition is unknown.

Reply:
Hmm... I think this was missed in the initial review. If a committed offset doesn't exist, it doesn't mean the topic or partition doesn't exist. For example, in the OffsetFetch API, we will return a null if there is no committed offset. For offset deletion, maybe we can find another error code or maybe treat it as a no-op and return no error? 

For reference, the existence check I was referring to in the OffsetCommit handler is this: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaApis.scala#L383. Before passing the partition to the coordinator, we explicitly check the metadata cache to see if we know about the topic.

Reply:
Yeah. That is right. I thought that the error would be appropriate when the group basically doesn’t know about the topic.

I will update the code the check the existence with the right approach, thanks for the hint.

If a topic exists but the group has not offsets about it, treating it as a no-op seems reasonable.

Reply:
I have opened a JIRA for this: https://issues.apache.org/jira/browse/KAFKA-8954. I will address it in a separate PR.